mitodl/mitxpro
# feat: add coupons deactivate form

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5435

### Description (What does it do?)
This PR adds the functionality to deactivate the coupons from ecommerce/admin UI

### Screenshots (if appropriate):

<img width="1152" alt="Screenshot 2024-09-26 at 2 55 22 PM" src="https://github.com/user-attachments/assets/cd466ff2-7dfc-4857-b921-95188c7d563c">

<img width="1153" alt="Screenshot 2024-10-10 at 1 13 31 PM" src="https://github.com/user-attachments/assets/6af76aa5-559e-4bc3-a93d-4f112a45235f">

<img width="1431" alt="Screenshot 2024-10-01 at 6 00 44 PM" src="https://github.com/user-attachments/assets/cc06952f-8387-4014-a6d0-4086961cbb12">

While submitting the button is disabled
<img width="1421" alt="Screenshot 2024-10-04 at 3 51 32 PM" src="https://github.com/user-attachments/assets/e3ac0a4a-03c7-4b8b-a782-dbe0c2e40adc">

<img width="1424" alt="Screenshot 2024-10-04 at 3 46 55 PM" src="https://github.com/user-attachments/assets/36dd80da-fe50-4738-9b80-22e6c40839cc">

If all the codes and names entered are invalid 
<img width="1424" alt="Screenshot 2024-10-04 at 3 47 20 PM" src="https://github.com/user-attachments/assets/cbb52525-2bcf-4bfc-a73d-6437c23e7113">




### How can this be tested?
- Go to http://localhost:8053/ecommerce/admin/deactivate-coupons/
- Enter the coupons names you want to deactivate. Each coupon at separate line. Submit the form
- Go to http://localhost:8053/admin/ecommerce/coupon/ and validate that the enable field of the coupons entered in the step  above is set to False
- Try to use the coupon in a checkout. You should get `Enrollment / Promotional Code 'coupon-code' is invalid` error

### Additional Context


- More details regarding access control: https://github.com/mitodl/hq/issues/5435#issuecomment-2400206915
